### PR TITLE
Add QuakeSpasm

### PR DESCRIPTION
**Pull request by erysdren**
Add popular Quake (1996) source-port modified to be comfortable on Haiku.

Big thanks to @OscarL for making the recipe workable.

Changes:
- Load mods and Quake game data from `B_USER_NONPACKAGED_DATA_DIRECTORY/QuakeSpasm/`
- Load engine pakfile from `B_SYSTEM_DATA_DIRECTORY/QuakeSpasm/`
- Add cross-compile scripts to build from Linux host
- Symlink to deskbar with logo added by @OscarL
- Tested and working on x86 and x86_x64 (x86_gcc2 not compatible)

### CODE REVIEW

File: games-fps/quakespasm/quakespasm-0.96.0~git.recipe [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+endian platforms.
+- OpenGL improvements. In-game resolution changing.
+- Widescreen FOV (field of view) support and automatic Hor+ FOV scaling.
+- Improved sound support. Support for OGG, MP3, Opus and WAV external music files.
+- Improved game controller support using SDL2.
+- Countless bug fixes, security fixes and other improvements.
+- Support for playing the 'Quake 2021 re-release' content"
+HOMEPAGE="https://quakespasm.sourceforge.net/"
+COPYRIGHT="
+	1996-2001 Id Software, Inc.
```
Comment (Begasus) [inline games-fps/quakespasm/quakespasm-0.96.0~git.recipe:19]: Start on the same line:
```diff
-COPYRIGHT="
-	1996-2001 Id Software, Inc.
+COPYRIGHT="1996-2001 Id Software, Inc.
```

File: games-fps/quakespasm/quakespasm-0.96.0~git.recipe [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	2002-2009 John Fitzgibbons and others
+	2010-2014 QuakeSpasm developers
+	"
```
Comment (Begasus) [inline games-fps/quakespasm/quakespasm-0.96.0~git.recipe:22]: Ending qoute on the same line:
```diff
-	2010-2014 QuakeSpasm developers
-	"
+	2010-2014 QuakeSpasm developers"
```

File: games-fps/quakespasm/quakespasm-0.96.0~git.recipe [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+SOURCE_DIR="quakespasm-$srcGitRev"
+ADDITIONAL_FILES="QuakeSpasm.rdef.in"
+PATCHES="quakespasm-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	quakespasm$secondaryArchSuffix = $portVersion
+	cmd:quakespasm$secondaryArchSuffix = $portVersion
```
Comment (Begasus) [inline games-fps/quakespasm/quakespasm-0.96.0~git.recipe:37]: The command is cp'd into $appsDir, didn't this trigger at the least a warning when packaging?
Should be: `app:QuakeSpasm` I think (didn't check yet).

**Comment by humdingerb** [comment]
Please add a quick note where to put any original game data etc. into the DESCRIPTION. That's a good idea in general for any games/apps that depend on stuff not included in the package.

File: games-fps/quakespasm/quakespasm-0.96.0~git.recipe [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libGL$secondaryArchSuffix
+	lib:libogg$secondaryArchSuffix
+	lib:libmad$secondaryArchSuffix
+	lib:libSDL2_2.0$secondaryArchSuffix
+	lib:libvorbis$secondaryArchSuffix
+	"
+
```
Comment (Begasus) [inline games-fps/quakespasm/quakespasm-0.96.0~git.recipe:55]: No empty line between PROVIDES and REQUIRES (same thing for BUILD_REQUIRES and BUILD_PREREQUIRES).

File: games-fps/quakespasm/patches/quakespasm-0.96.0~git.patchset [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+ endif
+ 
++ifeq ($(HOST_OS),haiku)
+ COMMON_LIBS:= -lGL
+-
+-ifeq ($(LINK_M),1)
+-COMMON_LIBS += -lm
++else
++COMMON_LIBS:= -lGL -lm
+ endif
```
Comment (Begasus) [inline games-fps/quakespasm/patches/quakespasm-0.96.0~git.patchset:58]: This shouldn't be needed?

File: games-fps/quakespasm/patches/quakespasm-0.96.0~git.patchset [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+ 
++ifeq ($(HOST_OS),haiku)
++QS_DATA_DIR=$(shell finddir B_USER_NONPACKAGED_DATA_DIRECTORY)/QuakeSpasm/
++QS_APP_DIR=$(shell finddir B_APPS_DIRECTORY)/QuakeSpasm/
++install:	quakespasm
++	mkdir -p $(QS_APP_DIR)
++	mkdir -p $(QS_DATA_DIR)
++	cp quakespasm $(QS_APP_DIR)
++	cp quakespasm.pak $(QS_DATA_DIR)
++else
```
Comment (Begasus) [inline games-fps/quakespasm/patches/quakespasm-0.96.0~git.patchset:73]: Needed? `make install` isn't used in the recipe?

File: games-fps/quakespasm/quakespasm-0.96.0~git.recipe [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+srcGitRev="70df2b661e9c632d04825b259e63ad58c29c01ac"
+SOURCE_URI="https://github.com/sezero/quakespasm/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="b89ba914d061376945b1fbe2cd0631e162f4f0d835e004eba87a0a9cce107ad4"
+SOURCE_DIR="quakespasm-$srcGitRev"
+ADDITIONAL_FILES="
+	QuakeSpasm.rdef.in
+	haiku_readme.txt
+	"
```
[the remark this answers is not in the document]
Comment (augiedoggie) [reply games-fps/quakespasm/quakespasm-0.96.0~git.recipe]: I think it's only inline when there is one file.  All of the other recipes I see start a new line when there are multiple files.

[the remark this answers is not in the document]
Comment (Begasus) [reply games-fps/quakespasm/quakespasm-0.96.0~git.recipe]: OK, no real example for this on the wiki, but I see others using it this way too ... sorry for the noise. :)

**Pull request merged**